src/Repository/RevisionRepository.php

```diff
@@ -48,6 +49,32 @@ public function findByContentType(ContentType $contentType, $orderBy = null, $li
             ], $orderBy, $limit, $offset);
     }
 
+    /**
+     * @param array<mixed> $search
+     */
+    public function search(array $search): Revisions
+    {
+        $qb = $this->createQueryBuilder('r');
+        $qb
+            ->join('r.contentType', 'c')
+            ->join('c.environment', 'e');
+
+        if (isset($search['contentType'])) {
+            $qb->andWhere($qb->expr()->eq('r.contentType', ':contentType'))->setParameter('contentType', $search['contentType']);
+        }
+        if (isset($search['modifiedBefore'])) {
+            $qb->andWhere($qb->expr()->lt('r.modified', ':modified'))->setParameter('modified', $search['modifiedBefore']);
+        }
+        if (isset($search['lockBy'])) {
+            $qb->andWhere($qb->expr()->eq('r.lockBy', ':lock_by'))->setParameter('lock_by', $search['lockBy']);
+        }
+        if (isset($search['archived'])) {
+            $qb->andWhere($qb->expr()->eq('r.archived', ':archived'))->setParameter('archived', $search['archived']);
+        }
+
+        return new Revisions($qb);
+    }
+
```

Comment:
I don't get it. why not a spefic searchFunciton searchRevision(?ContentType $contentTypee, ...). This is overriding the doctrine method and can introduce regression I'm afraid.

Reply:
There is no doctrine method "search", it is not overriding anything.